smartsim/_core/utils/telemetry/telemetry.py

```diff
@@ -355,20 +140,27 @@ def init_launcher(self, launcher: str) -> Launcher:
 
         raise ValueError("Launcher type not supported: " + launcher)
 
-    def set_launcher(self, launcher_type: str) -> None:
-        """Set the launcher for the experiment"""
-        self._launcher = self.init_launcher(launcher_type)
+    def set_launcher(self, launcher: str) -> None:
```

Comment:
Great catch. I'm splitting the job manager init stuff into it's own method to follow the same `init_xxx` pattern.

This also made me realize that I didn't need to return the launcher from `init_launcher` so that was tweaked to:

```py
    def init_launcher(self, launcher: str):
        """Initialize the controller with a specific type of launcher.
        SmartSim currently supports slurm, pbs(pro), lsf,
        and local launching

        :param launcher: the name of the workload manager used by the experiment
        :type launcher: str
        :raises SSUnsupportedError: if a string is passed that is not
        a supported launcher
        :raises TypeError: if no launcher argument is provided."""
        if not launcher:
            raise TypeError("Must provide a 'launcher' argument")

        if launcher_type := self._launcher_map.get(launcher.lower(), None):
            ### new here
            self._launcher = launcher_type()

        raise ValueError("Launcher type not supported: " + launcher)

    ### new here
    def init_job_manager(self) -> None:
        """Initialize the job manager instance"""
        if not self._launcher:
            raise TypeError("self._launcher must be initialized")

        self.job_manager.set_launcher(self._launcher)
        self.job_manager.start()

    def set_launcher(self, launcher: str) -> None:
        """Initialize all required dependencies

        :param launcher: the name of the workload manager used by the experiment
        :type launcher: str"""
        ### tweak here
        self.init_launcher(launcher)
        ### new here
        self.init_job_manager()
```

Reply:
lets go!